include/plugin_interface/SC_InlineBinaryOp.h

```diff
@@ -381,16 +382,16 @@ inline int sc_lcm(int a, int b)
 
 inline long sc_gcd(long a, long b)
 {
-	long t;
+	
+	if (a == 0) return b;
+	if (b == 0) return a;
 
+	long t;
```

Comment:
403-406 becomes

```c++
if (a < b) {
  long t = a;
  a = b;
  b = t;
}
while (b > 0) {
  long t = a % b;
  a = b;
  b = t;
}
```

there are three software engineering/C++ style ideas in this revision:
- declare variables only in the scope where they're used
- don't use a temp variable for two separate purposes
- give each statement its own line

i get that there is a certain appeal to not modifying code beyond the minimal editing needed. there is a discussion to be had about that. what i feel is that since we've pretty much all as a team acknowledged that this code is (1) extremely old and (2) not clearly written, we should take some extra effort to clean it up a bit while we're in there. i have gone overboard in some cases ( #2861 is an example of that) but if it's a matter of writing
```cpp
if (a == 0) return b;
```
instead of
```cpp
if (a == 0)
  return b;
```
when the first is what was done 20 years ago but the second is clearly better C++ style, i have to confess i don't see the point in sticking with the former. these are the kinds of things that i don't think we need a style guide to agree on, but maybe that is a faulty assumption since you and other contributors are not frequently working in C++. if so i am happy to go through that more transparent process.